src/common/playlist/playlist-manager.js

```diff
@@ -0,0 +1,115 @@
+// @flow
+import {KalturaPlayer} from '../../kaltura-player';
+import {FakeEvent, Utils} from 'playkit-js';
+import {PlaylistEventType} from './playlist-event-type';
+import getLogger from '../utils/logger';
+import {Playlist} from './playlist';
+import {PlaylistItem} from './playlist-item';
+
+class PlaylistManager {
+  static _logger: any = getLogger('PlaylistManager');
+
+  _player: KalturaPlayer;
+  _playlist: Playlist;
+  _options: KPPlaylistOptions;
+  _countdown: KPPlaylistCountdownOptions;
+
+  constructor(player: KalturaPlayer, options: KPOptionsObject) {
+    this._player = player;
+    this._playlist = new Playlist();
+    this._options = {autoContinue: true};
+    this._countdown = {duration: 10, showing: true};
+    this.configure(options.playlist);
+    this.addBindings();
+  }
+
+  configure(config: ?KPPlaylistConfigObject) {
+    if (config) {
+      this._playlist.configure(config);
+      Utils.Object.mergeDeep(this._options, config.options);
+      Utils.Object.mergeDeep(this._countdown, config.countdown);
+      if (this._playlist.items.find(item => !!item.sources)) {
```

Comment:
Maybe create `this._playlist.hasPlayableItems()` API for the playlist and do that logic inside the playlist itself?